Updating the minimum VS version

marcpopMSFT opened the pull request:
There have been issues with 8.0 running against 17.5

marcpopMSFT committed:
Updating the minimum VS version

There have been issues with 8.0 running against 17.5

marcpopMSFT commented:
Looks like I can't update this until 17.6 releases. I'll update the installer UI at least.

marcpopMSFT commented:
@rainersigwald looks like you added the read of the minimummsbuildversion file in SDK. Looks like I can't update the minimum version until after the version releases from now on. Is that about right? I think I'm ok with that but just thought I'd confirm that. https://github.com/dotnet/sdk/blob/main/eng/Versions.props#L109

rainersigwald commented:
I don't think that aspect of it was intentional. I think you could add the prerelease feed to unblock yourself for now:

https://github.com/dotnet/sdk/blob/f9b11911d4a0f53dcaba2c32f8b665d1f6ae2c4e/NuGet.config#L9

marcpopMSFT committed:
Add the prerelease msbuild feed

marcpopMSFT commented:
@rainersigwald build still failed. Did I get the right feed?

marcpopMSFT committed:
Update to the 17.6.1 build package as that's what's on the feed.
This forced an update to the config interop assembly and then collections immutable.

marcpopMSFT committed:
Merge branch 'main' into marcpopMSFT-updateminimummsbuild

marcpopMSFT commented:
@rainersigwald per your comment, overriding the build value with 17.6.1 which comes from the feed. This forced a few additional changes that I'll want to run by Daniel.

marcpopMSFT committed:
Add a comment, remove two of the references, and use a property for the other two.

marcpopMSFT committed:
Changed my mind and added back the other two. These are built for framework and we wouldn't find out till it was in VS if this weren't working.

marcpopMSFT commented on eng/Versions.props:
```diff
@@ -107,6 +107,7 @@
 
           This avoids the need to juggle references to packages that have been updated in newer MSBuild. -->
     <MicrosoftBuildPackageVersion Condition="exists('$(RepoRoot)\src\Layout\redist\minimumMSBuildVersion') and '$(DotNetBuildFromSource)' != 'true'">$([System.IO.File]::ReadAllText('$(RepoRoot)\src\Layout\redist\minimumMSBuildVersion').Trim())</MicrosoftBuildPackageVersion>
+    <MicrosoftBuildPackageVersion Condition="'$(MicrosoftBuildPackageVersion)' == '17.6.0'">17.6.1</MicrosoftBuildPackageVersion>
```
Added a comment and changed the references to use a property. I decided against removing the references as that's in the framework build and we may not know it's broken till we update the resolvers in VS in like August.
